src/NHSD.BuyingCatalogue.Organisations.Api/Validators/OrganisationValidator.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NHSD.BuyingCatalogue.Identity.Common.Models;
+using NHSD.BuyingCatalogue.Identity.Common.Results;
+using NHSD.BuyingCatalogue.Organisations.Api.Models;
+using NHSD.BuyingCatalogue.Organisations.Api.Repositories;
+
+namespace NHSD.BuyingCatalogue.Organisations.Api.Validators
+{
+    public sealed class OrganisationValidator : IOrganisationValidator
+    {
+        private readonly IOrganisationRepository _organisationRepository;
+
+        public OrganisationValidator(IOrganisationRepository organisationRepository)
+        {
+            _organisationRepository = organisationRepository ?? throw new ArgumentNullException(nameof(organisationRepository));
+        }
+
+        public async Task<Result> ValidateAsync(Organisation organisation)
+        {
+            if (organisation is null)
+                throw new ArgumentNullException(nameof(organisation));
+
+            var persistedOrganisation = await _organisationRepository.GetByOdsCodeAsync(organisation.OdsCode);
+
+            return persistedOrganisation is null
+                ? Result.Success()
+                : Result.Failure(new List<ErrorDetails> {new ErrorDetails("OrganisationAlreadyExists")});
```

Comment:
Although it is only one error at the moment, we might add more in the future. To encapsulate all the errors in one place for the organisation I would create an OrganisationErrors static class e.g. ApplicationUserErrors. This will allow re-use within the unit tests & encapsulate any parameters the error details required. However in this case there are no parameters.

Reply:
👍